version: "3.3"

services:
  fastapi-service:
    build: ./
    command: python3.11 manage.py  --api
    volumes:
      - ./:/usr/src/app
    ports:
      - "80:8000"
    restart: on-failure
    depends_on:
      - ocr-miner-psql
      - redis-cache

  ocr-miner-psql:
    image: postgres:alpine
    environment:
      POSTGRES_USER: myuser
      POSTGRES_PASSWORD: password
      POSTGRES_DB: ocr_miner_db
    ports:
      - "5432:5432"
    restart: on-failure
    volumes:
      - ./data/db:/var/lib/postgresql/data

  redis-cache:
    image: redis:alpine
    hostname: redis-cache
    container_name: redis-cache
    restart: on-failure
    ports:
      - '6379:6379'
    command: redis-server --requirepass eYVX7EwVmmxKPCDmwMtyKVge8oLd2t81
    volumes:
      - ./data/cache:/data
